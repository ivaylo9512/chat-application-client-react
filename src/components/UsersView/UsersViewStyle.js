import styled from "styled-components";

export const Container = styled.div`
    width: 100%;
    max-height: 100vh;
    overflow-y: scroll;
    padding: 8vh 0 3vh 0vh;
    height: 100%;
    display: flex;
    flex-direction: column;
`