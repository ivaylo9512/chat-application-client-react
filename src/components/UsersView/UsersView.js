import { Container } from './UsersViewStyle'
import UsersList from 'components/UsersList/UsersList';
import { getUsersQuery, usersRequest, resetUsersState, getUsersState, setCurrentUsers } from 'app/slices/usersSlice';
import Form from 'components/Form/Form';
import { useSelector, useDispatch } from 'react-redux';
import Pagination from 'components/Pagination/Pagination';
import { resetRequests } from 'app/slices/requestsSlice';
import { useEffect } from 'react';

const UsersView = () => {
    const dispatch = useDispatch();
    const { isLoading, error, dataInfo: { currentData } } = useSelector(getUsersState);

    useEffect(() => {
        if(!currentData){
            dispatch(resetRequests());
        }
    }, [currentData]);

    return(
        <>
            <Container>
                <UsersList />
                <Pagination selector={getUsersState} setData={setCurrentUsers} getData={usersRequest} pagesPerSlide={5}/>
                {error && <div data-testid='error'>{error}</div>}
            </Container>
            <Form action={usersRequest} resetState={resetUsersState} selector={getUsersQuery} placeholder={'Search users'} />

        </>
    )
}
export default UsersView