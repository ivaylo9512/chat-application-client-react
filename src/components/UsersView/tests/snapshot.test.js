import UsersView from 'components/UsersView/UsersView';
import { shallow } from 'enzyme';

describe('UserChatsView snapshot tests', () => {    
    const createWrapper = () => shallow(<UsersView />)
    
    it('should match snapshot', () => {
        const wrapper = createWrapper();

        expect(wrapper).toMatchSnapshot();
    })
})