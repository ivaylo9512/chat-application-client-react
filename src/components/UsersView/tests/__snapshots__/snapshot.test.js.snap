// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView snapshot tests should match snapshot 1`] = `
<Fragment>
  <styled.div>
    <UsersList />
    <Pagination
      getData={[Function]}
      pagesPerSlide={5}
      selector={[Function]}
      setData={[Function]}
    />
  </styled.div>
  <Form
    action={[Function]}
    placeholder="Search users"
    resetState={[Function]}
    selector={[Function]}
  />
</Fragment>
`;

exports[`UserView snapshot tests should match snapshot with error 1`] = `
<Fragment>
  <styled.div>
    <UsersList />
    <Pagination
      getData={[Function]}
      pagesPerSlide={5}
      selector={[Function]}
      setData={[Function]}
    />
    <div
      data-testid="error"
    >
      Unavailable
    </div>
  </styled.div>
  <Form
    action={[Function]}
    placeholder="Search users"
    resetState={[Function]}
    selector={[Function]}
  />
</Fragment>
`;

exports[`UserView snapshot tests should match snapshot with loading request 1`] = `
<Fragment>
  <styled.div>
    <UsersList />
    <Pagination
      getData={[Function]}
      pagesPerSlide={5}
      selector={[Function]}
      setData={[Function]}
    />
  </styled.div>
  <Form
    action={[Function]}
    placeholder="Search users"
    resetState={[Function]}
    selector={[Function]}
  />
</Fragment>
`;
